### DISCUSSION TITLE
Using X,Y,Z,Variable csv data to define elemental variables using PiecewiseConstantFromCSV

### DISCUSSION THREAD

**Discussion by Edward-Eth** [idaholab/moose#21037] [Q&A General] [answered]
I have a csv file of a value (T) as a function of x,y,z in a csv file, and was hoping to use the new PiecewiseConstantFromCSV to read that in such that each element is assigned an elemental value of T from the closest point defined in the csv file. At the moment this is working, except that it is assigning the variable to nodes instead of elements. Is there anyway to force an elemental value while using the nearest style reader?

**Comment by Edward-Eth** [comment, marked as answer]
Never-mind, figured it out. By making the variable an auxiliary variable and setting order = constant and family = monomial you can enforce elemental values from a nearest neighbour/voronoi reader. Not sure this is the best method available but it seems to work for me.

**Reply by GiudGiud** [reply]
hello

yes that's the way

**Reply by Edward-Eth** [reply]
Cool, thanks for the confirmation.
